### DISCUSSION TITLE
Getting a System.MissingMethodException: Method not found when trying to push to GitHub

### DISCUSSION THREAD

**Discussion by Rod-at-DOH** [community/community#140600] [Programming Help]
### Select Topic Area

Question

### Body

Recently I started getting this error message when trying to push my commits locally to our repo in GitHub using Azure Data Studio:

![image](https://github.com/user-attachments/assets/0d0db8a2-18d1-49d0-a750-5854a23147c1)

Looking at the Git log I see this:

> 2024-10-01 13:32:19.183 [info] Unhandled Exception: System.MissingMethodException: Method not found: 'Void System.CommandLine.Argument`1..ctor(System.String, System.String)'. at Microsoft.AzureRepos.AzureReposHostProvider.GitCredentialManager.Commands.ICommandProvider.CreateCommand() at GitCredentialManager.Application.RegisterProvider(IHostProvider provider, HostProviderPriority priority) at GitCredentialManager.Program.AppMain(Object o) at System.Threading.ThreadHelper.ThreadStart_Context(Object state) at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) at System.Threading.ThreadHelper.ThreadStart(Object obj) fatal: read error: Invalid argument fatal: expected flush after ref listing 2024-10-01 13:32:19.285 [info] > git config --get commit.template [97ms]

Why am I getting this? I would have thought that Azure Data Studio would work better than this. I'm working with version 1.49.1 on a Windows 11 Enterprise Edition PC.

**Comment by Topromm** [comment]
Could be due to an outdated version of GCM or some conflict?

Try to:
- **Update GCM**: Make sure you have the latest version of GCM installed. You can do that by either downloading the latest release from the [official GitHub repo](https://github.com/GitCredentialManager/git-credential-manager) or updating it directly via command line if you installed it that way.

- **Check Git version**: You should be able to update it either through Azure Data Studio or by installing the latest version from [git-scm.com](https://git-scm.com/).

- **Clear Git credentials**: Sometimes the GCM cache can cause weird issues. Try clearing your credentials and re-authenticating when you push again.

**Reply by Rod-at-DOH** [reply]
Hello @Topromm,

I'm running into problems trying to upgrade Git from 2.46 to 2.46.2. I've downloaded the upgrade, running it as Administrator. It insists that I install Visual Studio Code, again. (I've had it for years and it is up to date.) So, I've gone ahead and installed VS Code, again. Then I started the Git update to 2.46.2, again, and it will NOT let me proceed.

![image](https://github.com/user-attachments/assets/1dee486b-faea-4e0f-9fdd-5ef5b2f69614)

Suggestions, please.

**Reply by Rod-at-DOH** [reply]
Another question, what is the current version of GCM? Looking at the `main` branch it doesn't say anything about the current version. I've run this command `git credential-manager --version` and it gives me a version of

> 2.5.1+90d5f897f164a8fe07d2aa3bf7db439b0b4e03ea

on my system. Looking at the `release` branch I see "GCM 2.6", so I presume that is the current version. I'll try downloading that.

## Addendum
I just tried going to the latest installer for Windows version. I see 64-bit versions for Linux and macOS, but only 32-bit version for Windows. Is that correct? There's no 64-bit version for Windows?

**Reply by Topromm** [reply]
It’s odd that Git insists on reinstalling VSCode...

- Make sure all instances of VSCode are closed before running the Git installer.
- Try using a portable Git version instead of the standard installer. You can download it from [here](https://git-scm.com/download/win), and it shouldn't require admin rights or any dependencies like VS Code.

If neither of those worked, you can try manually uninstalling and then reinstalling Git from scratch. Sometimes a clean install resolves weird update issues.


GCM 2.6 is indeed the latest version, and it looks like you're currently running 2.5.1.
GCM for Windows only comes in a 32-bit version, but it works fine on 64-bit systems too.

**Reply by Rod-at-DOH** [reply]
I agree, Git's insistence that I must reinstall VSCode, even though I am at the current version, is odd.

I'll try installing the 32-bit version of GCM 2.6. 

@Topromm I went to download GCM 2.6. I was instructed that would be installed with [Git for Windows](https://github.com/git-ecosystem/git-credential-manager/blob/release/docs/install.md#windows), which I installed 3 days ago. So, I should already have GCM 2.6.

**Comment by mecodeatlas** [comment]
Hey @Rod-at-DOH! 👋 
We're going through our unanswered discussions, I know it has been a minute, but I wanted to check and see if you were able to get this solved?

**Reply by Rod-at-DOH** [reply]
Hi @mecodeatlas this is no longer an issue for me. In fact, I'd forgotten about it.
